prover/crates/bin/witness_generator/src/main.rs

```diff
@@ -103,22 +140,13 @@ async fn main() -> anyhow::Result<()> {
     let (stop_sender, stop_receiver) = watch::channel(false);
 
     let protocol_version = PROVER_PROTOCOL_SEMANTIC_VERSION;
-    let vk_commitments_in_db = match prover_connection_pool
-        .connection()
-        .await
-        .unwrap()
-        .fri_protocol_versions_dal()
-        .vk_commitments_for(protocol_version)
-        .await
-    {
-        Some(commitments) => commitments,
-        None => {
-            panic!(
-                "No vk commitments available in database for a protocol version {:?}.",
-                protocol_version
-            );
-        }
-    };
+    ensure_protocol_alignment(
```

Comment:
No strong opinion here. I just thought that a more defensive approach would be better. Imagine that for some reason protocol version for the existing batch changed in the DB -- we probably want to bail on that, even if it's not basic witness generator. Otherwise we will proceed and generate potentially incorrect witness.

But I'm OK with putting it to the BWG arm only if you think that's better.